lib/runtime/wasmer/exports_test.go

```diff
@@ -1210,3 +1210,138 @@ func TestInstance_TransactionPaymentCallApi_QueryCallFeeDetails(t *testing.T) {
 		require.Equal(t, test.expect, details)
 	}
 }
+
+func TestInstance_GrandpaGenerateKeyOwnershipProofEncoding(t *testing.T) {
+	t.Parallel()
+	keyOwnershipProof := types.OpaqueKeyOwnershipProof([]byte{64, 138, 252, 29, 127, 102, 189, 129, 207, 47, 157,
+		60, 17, 138, 194, 121, 139, 92, 176, 175, 224, 16, 185, 93, 175, 251, 224, 81, 209, 61, 0, 71})
+	encoded := scale.MustMarshal(keyOwnershipProof)
+	var proof types.OpaqueKeyOwnershipProof
+	err := scale.Unmarshal(encoded, &proof)
+	require.NoError(t, err)
+	require.Equal(t, keyOwnershipProof, proof)
```

Comment:
Shouldn't this test be in the `types` package? Since we are really testing `types.OpaqueKeyOwnershipProof` + `pkg/scale`

Reply:
good call!